web/olga/analytics/views.py

```diff
@@ -31,25 +32,39 @@ class AccessTokenRegistration(View):
     """
 
     @staticmethod
-    def create_edx_installation(access_token):
+    def get_or_create_access_token(uid):
         """
-        Create edx installation: insert access token field.
+        Provide access token for the given uid.
 
-        Access token is enough to fetching, sending and dispatching statistics. It called `Paranoid` level.
+        If uid already exist in database - return access token from storage,
+        otherwise create a new record with given uid and generated token.
+        :param uid: instance uid.
+        :return access_token.
         """
-        EdxInstallation.objects.create(access_token=access_token)
+        installation_data = EdxInstallation.objects.filter(uid=uid)
+        if installation_data.count() is 1:
+            access_token = installation_data[0].access_token
+            logger.debug('OLGA get previous edX installation with token %s for uid %s', access_token, uid)
+        else:
+            access_token = uuid4().hex
+            EdxInstallation.objects.create(access_token=access_token, uid=uid)
+            logger.debug('OLGA registered edX installation with token %s for uid %s', access_token, uid)
+        return access_token
 
     def post(self, request):  # pylint: disable=unused-argument
         """
         Receive primary edX installation request for getting access for dispatch statistics via token.
 
         Returns HTTP-response with status 201, that means object (installation token) was successfully created.
         """
-        access_token = uuid4().hex
-        self.create_edx_installation(access_token)
+        uid_serializer = UidForm(request.POST)
+        uid = str(request.POST.get('uid'))
```

Comment:
`uid` should be taken only if the form is valid, cleaned_data['uid'] would be an appropriate method for this purpose.